### ISSUE TITLE
push_tasks_from_* queries should use pagination

### ISSUE THREAD

**Issue by marco-c** [mozilla/mozci#11]
At least in the unittest case, we can miss things because we have more than 10000 results.

For example, when analyzing `d215b3534e043455cad3bf14a90e5f80e9f8cdd0`, I was missing the unittest result for the task `S-1IQ42WTNuRZs_1a691dw`.

By adding `- eq: {task.id: "S-1IQ42WTNuRZs_1a691dw"}` to the query, I got it.

**Comment by marco-c** [comment]
For the pagination we could:
1) run a first query where we count the results, grouping by task name;
2) using the task names and their result counts from 1, form groups of task names so that each group contains at maximum 10000 results;
3) perform a query for each 10000 group, using `{"in": {"task.name": group_of_task_names}}`

@ahal @klahnakoski do you see any other solution?

**Comment by klahnakoski** [comment]
@marco-c   Yes, that is what I do, assuming each group is small enough to be under the 10K limit, I accumulate the groups using large `group_of_task_names`

I have also let the machine explore the groups based on `prefix`, and then make more calls if there are too many groups for a given prefix:

First, find out how many groups, by prefix

```
{
	"from":"task",
	"select":{"value":"task.label","aggregate":"cardinality"},
	"groupby":{"name":"prefix","value":{"left":{"task.label":1}}},  //PREFIX IS 1
	"limit":10000
}
```

for groups that are too large, increase the length of the prefix and try again:

```
{
	"from":"task",
	"select":{"value":"task.label","aggregate":"cardinality"},
	"groupby":{"name":"prefix","value":{"left":{"task.label":2}}},  //PREFIX IS 2
	"where":{"prefix":{"task.label":"t"}},
	"limit":10000
}
```

**Comment by marco-c** [comment]
Or we can use the new feature @klahnakoski built: https://github.com/mozilla/ActiveData/pull/166.

**Comment by marco-c** [comment]
Depends on https://github.com/mozilla/adr/issues/45.

**Comment by marco-c** [comment]
I guess this is kind of fixed by #21. Infact, we could now remove the pagination we have for push_revisions.
